locations/spiders/gucci.py

```diff
@@ -1,84 +1,59 @@
+        "https://www.gucci.com/it/it/",
+        "https://www.gucci.com/jp/ja/",
+        "https://www.gucci.com/kr/ko/",
+        "https://www.gucci.com/kw/en_gb/",
+        "https://www.gucci.com/nl/en_gb/",
+        "https://www.gucci.com/no/en_gb/",
+        "https://www.gucci.com/nz/en_au/",
+        "https://www.gucci.com/pl/en_gb/",
+        "https://www.gucci.com/pt/en_gb/",
+        "https://www.gucci.com/qa/en_gb/",
+        "https://www.gucci.com/ro/en_gb/",
+        "https://www.gucci.com/sa/en_gb/",
+        "https://www.gucci.com/se/en_gb/",
+        "https://www.gucci.com/sg/en_gb/",
+        "https://www.gucci.com/si/en_gb/",
+        "https://www.gucci.com/tr/en_gb/",
+        "https://www.gucci.com/uk/en_gb/",
+        "https://www.gucci.com/us/en/",
+    ]
+
+    def start_requests(self):
+        for url in self.sites:
+            locale = url.split("/")[-2]
+            sitemap_url = url + "sitemap/STORE-{}.xml".format(locale)
+            yield scrapy.Request(sitemap_url, self._parse_sitemap)
 
     def parse(self, response):
-        data = response.json()
-        stores = data["features"]
-
-        for store in stores:
-            ## City tag contains city, state, postal, country
-            addr = store["properties"]["address"]["city"].split(",")
-            city = addr[0].strip()
-            if len(addr) == 3:
-                state = ""
-                postal = addr[1].strip()
-            elif len(addr) == 4:
-                state = addr[1].replace("Canada", "").strip()
-                postal = addr[2].strip()
-            else:
-                state = ""
-                postal = ""
-
-            properties = {
-                "ref": store["properties"]["url"],
-                "name": store["properties"]["name"],
-                "addr_full": store["properties"]["address"]["location"],
-                "city": city,
-                "state": state,
-                "postcode": postal,
-                "country": store["properties"]["address"]["country"],
-                "lat": store["properties"]["latitude"],
-                "lon": store["properties"]["longitude"],
-                "phone": store["properties"]["address"]["phone"],
-                "website": response.urljoin(store["properties"]["url"]),
-                "extras": {"number": store["properties"]["storeCode"]},
-            }
-
-            try:
-                selector = scrapy.Selector(
-                    text=store["properties"]["openingHours"]["h24"], type="html"
-                )
-                days = selector.xpath('//*[@class="day"]/text()').extract()
-                hours = selector.xpath('//*[@class="all"]/text()').extract()
-
-                hours = self.parse_hours(days, hours)
-
-                if hours:
-                    properties["opening_hours"] = hours
-
-            except:
-                pass
-
-            yield GeojsonPointItem(**properties)
+        yield LinkedDataParser.parse(response, "Store")
```

Comment:
Looks like they don't include lat/lon in the linked data JSON, but they do include it in the HTML at `section[@id="store-locator"]`

Reply:
Yep, some pipeline steps that insert statistics about missing or invalid important (but not required) fields like lat/lon along with format checks on opening_hours and website would be helpful.